src/LinuxTracing/SwitchesStatesNamesVisitorTest.cpp

```diff
@@ -0,0 +1,498 @@
+  static constexpr uint64_t kInTimestampNs1 = 111;
+  static constexpr uint64_t kOutTimestampNs1 = 112;
+  static constexpr uint64_t kNewTimestampNs2 = 120;
+  static constexpr uint64_t kInTimestampNs2 = 121;
+  static constexpr uint64_t kOutTimestampNs2 = 122;
+  static constexpr uint64_t kWakeTimestampNs2 = 123;
+  static constexpr uint64_t kStopTimestampNs = 130;
+  static constexpr int64_t kRunnableStateMask = 0;
+  static constexpr int64_t kInterruptibleSleepStateMask = 0x01;
+  static constexpr int64_t kDeadStateMask = 0x10;
+
+  void ProcessFakeEventsForThreadStateTests();
+};
+
+std::unique_ptr<ForkPerfEvent> MakeFakeForkPerfEvent(pid_t pid, pid_t tid) {
+  auto event = std::make_unique<ForkPerfEvent>();
+  event->ring_buffer_record.pid = pid;
+  CHECK(event->GetPid() == pid);
+  event->ring_buffer_record.tid = tid;
+  CHECK(event->GetTid() == tid);
+  return event;
+}
+
+std::unique_ptr<ExitPerfEvent> MakeFakeExitPerfEvent(pid_t pid, pid_t tid) {
+  auto event = std::make_unique<ExitPerfEvent>();
+  event->ring_buffer_record.pid = pid;
+  CHECK(event->GetPid() == pid);
+  event->ring_buffer_record.tid = tid;
+  CHECK(event->GetTid() == tid);
+  return event;
+}
+
+std::unique_ptr<TaskNewtaskPerfEvent> MakeFakeTaskNewtaskPerfEvent(pid_t new_tid, const char* comm,
+                                                                   uint64_t timestamp_ns) {
+  auto event = std::make_unique<TaskNewtaskPerfEvent>();
+  event->ring_buffer_record.data.pid = new_tid;
+  CHECK(event->GetNewTid() == new_tid);
+  CHECK(strlen(comm) < 16);
+  strncpy(event->ring_buffer_record.data.comm, comm, 16);
+  CHECK(strcmp(event->GetComm(), comm) == 0);
+  event->ring_buffer_record.sample_id.time = timestamp_ns;
+  CHECK(event->GetTimestamp() == timestamp_ns);
+  return event;
+}
+
+std::unique_ptr<TaskRenamePerfEvent> MakeFakeTaskRenamePerfEvent(pid_t renamed_tid,
+                                                                 const char* new_comm,
+                                                                 uint64_t timestamp_ns) {
+  auto event = std::make_unique<TaskRenamePerfEvent>();
+  event->ring_buffer_record.data.pid = renamed_tid;
+  CHECK(event->GetRenamedTid() == renamed_tid);
+  CHECK(strlen(new_comm) < 16);
+  strncpy(event->ring_buffer_record.data.newcomm, new_comm, 16);
+  CHECK(strcmp(event->GetNewComm(), new_comm) == 0);
+  event->ring_buffer_record.sample_id.time = timestamp_ns;
+  CHECK(event->GetTimestamp() == timestamp_ns);
+  return event;
+}
+
+std::unique_ptr<SchedSwitchPerfEvent> MakeFakeSchedSwitchPerfEvent(
+    uint32_t cpu, pid_t prev_pid_or_minus_one, pid_t prev_tid, int64_t prev_state_mask,
+    pid_t next_tid, uint64_t timestamp_ns) {
+  auto event = std::make_unique<SchedSwitchPerfEvent>();
+  event->ring_buffer_record.sample_id.cpu = cpu;
+  CHECK(event->GetCpu() == cpu);
+  event->ring_buffer_record.sample_id.pid = prev_pid_or_minus_one;
+  CHECK(event->GetPrevPidOrMinusOne() == prev_pid_or_minus_one);
```

Comment:
should these checks be a test by themselves? I kind of feel they do not belong here. and if the tests are ordered so that this comes before the actual test we will get crash with hard to read result.

Reply:
Imagine a situation where you have a test that checks this. It is a separate test.
Imagine then that I broke this functionality. Now I am executing the tests and this check gets executed first. In this case
the test that is actually ASSERTing this is never executed and if you look in list of failed test cases it is misleading.  In the case of XML based result on runner you may even get no results at all and need to execute it locally to get to the bottom of this.

What I am saying is this check does not belong here. It is an assert that belongs to a different test. And this test shouldn't interfere with that other test.

Reply:
It's not functionality, it's test code. The assertions are there for extra safety that the setup of the tests is correct, but are technically redundant. If you are referring to testing the specific methods of the single `PerfEvent` classes, those could be added somewhere else, but it's out of scope here.

Reply:
well, if it is out of scope, maybe remove the CHECKs?

Reply:
But why? How is less better here? They verify that the fake `PerfEvent`s are being set up correctly, hence that the premises of the tests are correct.
They are similar to the ones in `GpuTracepointVisitorTest` and `LostAndDiscardedEventVisitorTest`.

Reply:
Maybe you're missing that the point is not to verify the methods (e.g., `GetCpu()`), but that the assignment to `ring_buffer_record` was correct?

Reply:
Ok, so... I do not understand. If we want to verify this - why we are doing it here and not in another unit-test with asserts? If not why this CHECK is here?

Reply:
We are not testing production code, but just asserting that this piece of purely test code (the assignments) is correct. The affected tests should indeed just not run if those assignments are wrong.

Reply:
If your underlying argument is that we should not have `CHECK`s in tests at all then I simply disagree. We have them in various places, `CHECK`s for the behavior of the test code, `EXPECT` and `ASSERT` for the actual code under test. A very clear distinction too.